reframe/core/schedulers/slurm.py

```diff
@@ -91,8 +91,35 @@ def __init__(self):
         self._update_state_count = 0
         self._completion_time = None
 
-    @property
-    def completion_time(self):
+    def completion_time(self, job):
+        if (self._completion_time or
+            not slurm_state_completed(job.state)):
+            return self._completion_time
+
+        slurm_variable = [('SLURM_TIME_FORMAT', 'standard')]
+        with env.temp_environment(variables=slurm_variable):
+            completed = _run_strict(
+                'sacct -S %s -P -j %s -o jobid,end' %
+                (datetime.now().strftime('%F'), job.jobid),
+                log=False
+            )
+
+        # This matches the format for both normal jobs as well as job arrays.
+        # For job arrays the job_id has one of the following formats:
+        #   * <job_id>_<array_task_id>
+        #   * <job_id>_[<array_task_id_start>-<array_task_id_end>]
+        # See (`Job Array Support<https://slurm.schedmd.com/job_array.html`__)
+        state_match = list(re.finditer(
```

Comment:
`re.findall()` should be equivalent to `list(re.finditer())`. Please update it in `_update_state()` as well.

Reply:
It is not equivalent. `re.findall()` returns a list of tuples and I won't be able to use named variables anymore, like `s.group('state')`. I would have to access the tuple elements with their index.

Reply:
Ok, that's fine. I think that was the reason that we have used this construct then.